desktop/core/src/desktop/api_public.py

```diff
@@ -162,6 +164,14 @@ def storage_upload_file(request):
   django_request = get_django_request(request)
   return filebrowser_views.upload_file(django_request)
 
+# Impala
+
+@api_view(["POST"])
+def analyze_table(request, dialect, database, table, columns=None):
+  django_request = get_django_request(request)
+
+  if dialect in ["impala", "beeswax"]:
```

Comment:
@romainr Whats the best way to the case when this API is not for some dialects?

For eg.. `https://demo.gethue.com/api/mysql/analyze/db/table` would currently give the following error with `500 Internal Server Error`

AssertionError: Expected a `Response`, `HttpResponse` or `HttpStreamingResponse` to be returned from the view, but received a `<class 'NoneType'>`

Reply:
It is not supported by all the dialects yet, in the meantime doing what they say is fine, e.g. returning an empty HttpResonse() and not None